src/EventStore.Client/EventStoreClientBase.cs

```diff
@@ -44,6 +50,10 @@ public abstract class EventStoreClientBase :
 			_exceptionMap = exceptionMap;
 			_cts = new CancellationTokenSource();
 			_channelCache = new(Settings);
+			_httpClient = new Lazy<HttpClient>(CreateHttpClient);
```

Comment:
perhaps we could have a `HttpFallback` class that encapsulates the http stuff to avoid it complicating the ClientBase itself. Construct it in the EventStoreClientBase constructor (passing in the Settings) and forward the HttpGet and HttpPost methods to it?

Reply:
perhaps append `Async` to the async method names also

Reply:
in fact the ChannelInfo can be passed into HttpGet(Async) from the place that is doing the fallback (e.g. ListAllAsync) since it has already called GetChannelInfo

Reply:
Done